From 99375355644f3f2f6089f05a8cfe6d8135e23639 Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Thu, 15 Mar 2012 15:00:00 +0900
Subject: [PATCH 06/97] ext/dfbvideosink: ignore alpha channel in source data

This is a temporal workaround for ARGB8888 data output by VIN.
---
 ext/directfb/dfbvideosink.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index 8a26356..1409a7a 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1726,6 +1726,8 @@ gst_dfbvideosink_show_frame (GstBaseSink * bsink, GstBuffer * buf)
     viosurface[0].bpitchc = 0;
     viosurface[0].bpitcha = 0;
     viosurface[0].format = dfb2shvio_format (src_format);
+    if (viosurface[0].format == REN_ARGB32)
+      viosurface[0].format = REN_XRGB32;        /* WORKAROUND: ignore the alpha channel */
     if (viosurface[0].format == REN_UNKNOWN) {
       GST_WARNING_OBJECT (dfbvideosink, "Pixel format %s unsupported",
           gst_dfbvideosink_get_format_name (src_format));
-- 
1.7.10.4

